models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatModel.java

```diff
@@ -499,6 +500,10 @@ GeminiRequest createGeminiRequest(Prompt prompt, VertexAiGeminiChatOptions updat
 			generativeModelBuilder.setTools(tools);
 		}
 
+		if (prompt.getOptions() instanceof VertexAiGeminiChatOptions options && options.getSafetySettings() != null) {
```

Comment:
You have to check for `!CollectionUtils.isEmpty(options.getSafetySettings())` instead. 
Since the `options.getSafetySettings()` would never be null (you are creating empty set by default) it will always override the default SafetySettings. 
It is anther question as how to set the SafetySettings via the `spring.ai.vertex.ai.gemini.chat.options.SafetySettings`?